src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs

```diff
@@ -343,7 +350,88 @@ public void Stop()
             }
         }
 
+        /* W3C support functionality (see https://w3c.github.io/trace-context) */
+
+        /// <summary>
+        /// Holds the W3C 'tracestate' header.   This is typically used by logging systems
+        /// to store vendor-specific information that must flow.
+        /// Logically it is just a kind of baggage (if flows just like baggage), but because
+        /// it is expected to be special cased (it has its own HTTP header), it is more 
+        /// convenient/efficient if it is not lumped in with other baggage.   
+        /// </summary>
+        public string TraceState
+        {
+            get
+            {
+                for (var activity = this; activity != null; activity = activity.Parent)
```

Comment:
Add a check for an infinite loop.

Reply:
After reviewing the code I believe that it is impossible for the parent chain to have a loop.  I will add a comment as to the reasoning.